src/storage/src/render/upsert.rs

```diff
@@ -41,12 +43,15 @@ use timely::progress::{Antichain, Timestamp};
 use crate::healthcheck::HealthStatusUpdate;
 use crate::metrics::upsert::UpsertMetrics;
 use crate::render::sources::OutputIndex;
+use crate::render::upsert::autospill::{AutoSpillBackend, RocksDBParams};
+use crate::render::upsert::memory::InMemoryHashMap;
 use crate::render::upsert::types::{
-    upsert_bincode_opts, AutoSpillBackend, InMemoryHashMap, RocksDBParams, UpsertState,
-    UpsertStateBackend,
+    upsert_bincode_opts, StateValue, UpsertState, UpsertStateBackend, Value,
 };
 use crate::storage_state::StorageInstanceContext;
 
+mod autospill;
+mod memory;
```

Comment:
+1 good to split them up into separate modules. The types file was already too big

Reply:
i know!